net/gateway/gateway.go

```diff
@@ -0,0 +1,157 @@
+package gateway
+
+import (
+	"fmt"
+	"go.sia.tech/core/net"
+	"go.sia.tech/core/net/mux"
+	"go.sia.tech/core/types"
+	"io"
+	gnet "net"
```

Comment:
ehh yeah having to rename a stdlib package kinda sucks. Might be ok if it's rare to import both `net` packages, I guess? But this is definitely evidence against my suggestion.